### PR TITLE
Add AGC support for USRP Source block.

### LINKED ISSUE #2442: gr-uhd AGC support
UHD currently has the ability to turn AGC on/off via this function: https://files.ettus.com/manual/classuhd_1_1usrp_1_1multi__usrp.html#abdab1f6c3775a9071b15c9805f866486

I am hoping that this functionality gets exposed in gr-uhd, so that I can toggle AGC on/off using GNU Radio.

### PR DESCRIPTION
**Pull request by dwd-pete**
Fixes #2442

### CODE REVIEW

**Comment by dwd-pete** [comment]
Full disclosure: I don't have a USRP to test with, but the generated GRC blocks look sane as does the generated top_block.py for each possible set of gain options.

**Comment by dkozel** [comment]
@dwd-pete Thanks for posting this up. I've quickly read over and everything makes logical sense and I can't think of edge cases you've missed. I do have a B2xx USRP so can do a functional test early next week if not sooner.

**Comment by marcusmueller** [comment]
Hi!

I agree with @dkozel, this looks nice, @dwd-pete! Seeing this is not just a handful lines of code, and you're a first-time contributor: Do we already have a copyright assignment from you (or the organization you work for)? If not, @bhilburn can set you up with one.

**Comment by dwd-pete** [comment]
I currently do not have a CLA in place, and totally forgot about this. That said, I have gotten the ball rolling between my organization and @bhilburn  to get the CLA taken care of.

**Comment by dwd-pete** [comment]
CLA should now be in place.

File: gr-uhd/grc/gen_uhd_usrp_blocks.py [hunk 1]
```diff
@@ -202,7 +217,14 @@
     -   set_samp_rate(${'$'}{samp_rate})
     % for n in range(max_nchan):
     -   set_center_freq(${'$'}{${'center_freq' + str(n)}}, ${n})
+    % if sourk == 'source':
+    -   ${'$'}{'set_rx_agc(True, ${n})' if context.get('rx_agc${n}')() == 'Enabled' else ''}
```
Comment (mbr0wn) [inline gr-uhd/grc/gen_uhd_usrp_blocks.py]: I wonder if we can't get those True/False values directly from the rx_agc var, but this code is touchy and anything that works is fine.

File: gr-uhd/lib/usrp_source_impl.cc [hunk 2]
```diff
@@ -159,6 +159,17 @@ namespace gr {
       return _dev->set_rx_gain(gain, name, chan);
     }
 
+    void
+    usrp_source_impl::set_rx_agc(const bool enable, size_t chan)
+    {
+#if UHD_VERSION >= 30900
```
Comment (mbr0wn) [inline gr-uhd/lib/usrp_source_impl.cc]: It's 30803, actually.

**Comment by mbr0wn** [comment]
Merged manually.
